pkg/jsonstream/image_progress.go

```diff
@@ -0,0 +1,47 @@
+package jsonstream
+
+import (
+	"fmt"
+
+	"github.com/containerd/containerd/progress"
+)
+
+const (
+	// PullStatusDownloading represents downloading status.
+	PullStatusDownloading = "downloading"
+	// PullStatusWaiting represents waiting status.
+	PullStatusWaiting = "waiting"
+	// PullStatusResolving represents resolving status.
+	PullStatusResolving = "resolving"
+	// PullStatusResolved represents resolved status.
+	PullStatusResolved = "resolved"
+	// PullStatusExists represents exist status.
+	PullStatusExists = "exists"
+	// PullStatusDone represents done status.
+	PullStatusDone = "done"
+)
```

Comment:
Maybe define a special type for these status? like `type PullStatus string` ?

Reply:
It's const var. I think both are ok right now. If we need to upgrade the definition with format like `progress.Bar`, we will use the specific type for the status. WDTY?

Reply:
Yes, agree.